include/envoy/network/transport_socket.h

```diff
@@ -149,9 +149,12 @@ class TransportSocketFactory {
   virtual bool implementsSecureTransport() const PURE;
 
   /**
+   * @param overrideServerNameIndication set SNI, disregard the value the factory was configured
```

Comment:
avoid directly refer SNI in transport socket interface, this is not specific to TLS. more context: https://github.com/envoyproxy/envoy/pull/4662#discussion_r223895517